apps/meteor/client/views/hooks/roomActions/useDeleteRoom.tsx

```diff
@@ -13,13 +13,23 @@ export const useDeleteRoom = (room: IRoom | Pick<IRoom, RoomAdminFieldsType>, {
 	const router = useRouter();
 	const setModal = useSetModal();
 	const dispatchToastMessage = useToastMessageDispatch();
-	const hasPermissionToDelete = usePermission(`delete-${room.t}`, room._id);
-	const canDeleteRoom = isRoomFederated(room) ? false : hasPermissionToDelete;
 
 	const isAdminRoute = router.getRouteName() === 'admin-rooms';
 
 	const deleteRoomEndpoint = useEndpoint('POST', '/v1/rooms.delete');
 	const deleteTeamEndpoint = useEndpoint('POST', '/v1/teams.delete');
+	const teamsInfoEndpoint = useEndpoint('GET', '/v1/teams.info');
+
+	const teamId = room.teamId || '';
+	const { data: teamInfoData } = useQuery(['teamId', teamId], async () => teamsInfoEndpoint({ teamId }), {
+		keepPreviousData: true,
+		retry: false,
+		enabled: room.teamId !== '',
+	});
+
+	const hasPermissionToDeleteRoom = usePermission(`delete-${room.t}`, room._id);
+	const hasPermissionToDeleteTeamRoom = usePermission(`delete-team-${room.t === 'c' ? 'channel' : 'group'}`, teamInfoData?.teamInfo.roomId);
+	const canDeleteRoom = isRoomFederated(room) ? false : hasPermissionToDeleteRoom && hasPermissionToDeleteTeamRoom;
```

Comment:
In this case, we still need the global delete room permission, is this the expected outcome?

Also, what happens if the room is not part of a team, will `hasPermissionToDeleteTeamRoom` be always false?

Reply:
Oops, we just need it if it's in a team room
~That's what happens when you test things only with the admin user~
Fixed!